### PR TITLE
ELPP-3278 Draft of Node major/minor versions ADR

### PR DESCRIPTION
**Pull request by davidcmoulton**
This should work for eLife 2 stack, and future new internal projects.

Collaborations with 3rd parties may require a different set of rules.

### CODE REVIEW

File: adr/0006-supported-node-and-npm-versions.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Node 8.9.3 includes bundled npm version 5.5.1. We will use the npm version bundled with Node.
+
+When upgrading, we will make a concerted effort to upgrade all projects as part of the same piece of work.  
+
+
+## Consequences
+
+New projects should use Node 8.9.3 and npm 5.5.1.
+
+Exiting projects should be upgraded to use 8.9.3 before April 2018 (when Active LTS ends for the 6.x line ends).
```
Comment (seanwiseman) [inline adr/0006-supported-node-and-npm-versions.md]: small typo 'exiting'

**Review by thewilkybarkid — COMMENTED** [review]
We definitely want to avoid patch numbers in the ADR, but I think we could make this more generic and avoid versions entirely: new projects always use latest LTS, existing projects upgraded to latest LTS by the time their version is inactive.

File: adr/0006-supported-node-and-npm-versions.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+Staying up to date with the major and minor versions of Node is important:
+
+- to be able to use new features (nice to have)
+- to keep working on a version that receives bug fixes and security updates (must have)            
+
+## Decision
+
+To use the latest Long Term Support (LTS) version.
+
+The 8.x is the current Node LTS line. The current Node LTS version is 8.9.3.
```
Comment (giorgiosironi) [inline adr/0006-supported-node-and-npm-versions.md]: I agree on including `8.x` as the actual `LTS` changes as time passes, and we should revise the decision after that has changed. I think minor and patch versions are too detailed for an ADR instead, their place is in formulas and similar automation configuration.

File: adr/0006-supported-node-and-npm-versions.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+In order to provide dependency management consistency, we need a consensus on which npm versions to support.
+
+Staying up to date with the major and minor versions of Node is important:
+
+- to be able to use new features (nice to have)
+- to keep working on a version that receives bug fixes and security updates (must have)            
+
+## Decision
+
+To use 8.x, the current Long Term Support (LTS) line.
```
Comment (giorgiosironi) [inline adr/0006-supported-node-and-npm-versions.md:26]: @elifesciences/yld will this be enough also to run and contribute to xpub releases? From an operations point of view, the intention is to run everything as a container image, but local development is influenced by these choices.

Comment (tamlyn) [reply adr/0006-supported-node-and-npm-versions.md:26]: Yes, we're targeting the latest LTS release (currently 8.9).

**Review by thewilkybarkid — APPROVED** [review]

**Review by giorgiosironi — APPROVED** [review]

**Review by seanwiseman — APPROVED** [review]

File: adr/0006-supported-node-and-npm-versions.md (hunk 3 above)
Comment (davidcmoulton) [inline adr/0006-supported-node-and-npm-versions.md:26]: @elifesciences/article-viewer Will this be compatible with eLife's Lens?

Comment (gnott) [reply adr/0006-supported-node-and-npm-versions.md:26]: It looks like I may have used v6.10.3 the last time I built Lens. I will test 8.x and see how it goes.

Comment (gnott) [reply adr/0006-supported-node-and-npm-versions.md:26]: I successfully built Lens using node v8.9.4 today.

Comment (giorgiosironi) [reply adr/0006-supported-node-and-npm-versions.md:26]: Do we have your approval then?

Comment (davidcmoulton) [reply adr/0006-supported-node-and-npm-versions.md:26]: Taking the successful build as approval. @gnott shout up if there's anything that argues against this and we can review.

**Pull request merged**